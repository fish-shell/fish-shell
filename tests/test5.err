No matches for wildcard '*ee*'.
fish: 	case *ee*
      	     ^
