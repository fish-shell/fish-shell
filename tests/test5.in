
set smurf green

switch $smurf;
	case "*ee*"
		echo Test 1 pass
	case "*"
		echo Test 1 fail
end;

switch $smurf
	case *ee*
		echo Test 2 fail
	case red green blue
		echo Test 2 pass
	case "*"
		echo Test 2 fail
end

switch $smurf
	case cyan magenta yellow
		echo Test 3 fail
	case "?????"
		echo Test 3 pass
end

# Verify that we can do wildcard expansion when we
# don't have read access to some path components
# See #2099
set -l where ../test/temp/fish_wildcard_permissions_test/noaccess/yesaccess
mkdir -p $where
chmod 300 (dirname $where) # no read permissions
mkdir -p $where
touch $where/alpha.txt $where/beta.txt $where/delta.txt
echo $where/*
chmod 700 (dirname $where) # so we can delete it
rm -rf ../test/temp/fish_wildcard_permissions_test
