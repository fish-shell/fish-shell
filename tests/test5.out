Test 1 pass
Test 2 pass
Test 3 pass
../test/temp/fish_wildcard_permissions_test/noaccess/yesaccess/alpha.txt ../test/temp/fish_wildcard_permissions_test/noaccess/yesaccess/beta.txt ../test/temp/fish_wildcard_permissions_test/noaccess/yesaccess/delta.txt
